src/client/pages/Geo/GeoMap/GeoMapMenuStatistics/StatisticalGraphsPanel/StatisticalGraphsPanel.tsx

```diff
@@ -0,0 +1,172 @@
+    // Extract the labels from the first column of the data.
+    return row[0]
+  })
+  labels.push(`Reported to FRA ${year}`) // Label for fra1aForestArea
+
+  const areas = data.map((row) => {
+    // Extract the areas from the second column of the data.
+    const area = row[1]
+    return area
+  })
+  areas.push(fra1aForestArea * 1000) // Convert to Ha. Instead of Thousands of Ha.
+  const maximumArea = Math.max(...areas)
+
+  const percentages = data.map((row) => {
+    // Extract the percentages from the third column of the data.
+    const area = row[2]
+    return area
+  })
+  const fra1aForestAreaPercentage = (fra1aForestArea * 1000 * 100) / (fra1ALandArea * 1000)
+  percentages.push(Number(fra1aForestAreaPercentage.toFixed(2))) // Percentage for fra1aForestArea.
+
+  const backgroundColors = data.map((row) => {
+    const source = row[0] as ForestSource
+    if (source.toUpperCase().indexOf('HANSEN') === -1) {
+      return sourcesMetadata[source].palette[0]
+    }
+    return sourcesMetadata.Hansen.palette[0]
+  })
+  backgroundColors.push('#000000') // Black color for the fra1aForestArea bar.
+
+  const chartData = {
+    labels,
+    datasets: [
+      {
+        label: 'Area',
+        backgroundColor: backgroundColors,
+        data: areas,
+        yAxisID: 'Areas',
+      },
+    ],
+  }
+
+  const options = {
+    plugins: {
+      title: {
+        display: true,
+        text: chartTitle,
+      },
+      legend: {
+        display: false,
+      },
+    },
+    scales: {
+      Areas: {
+        type: 'linear',
+        display: true,
+        position: 'left',
+        suggestedMax: maximumArea * (1 + 0.2), // Add 20 % buffer area to the top of the bars
+        ticks: {
+          callback(value: number) {
+            return `${value / 1000000} Million`
+          },
+        },
+      },
+    },
+  } as unknown as ChartOptions<'bar'>
+
+  // Pluging to get a white background color when downloading
+  const plugins: Plugin[] = [
+    {
+      id: 'white-background-on-download',
+      beforeDraw: (chartCtx) => {
+        const ctx = chartCtx.canvas.getContext('2d')
+        ctx.save()
+        ctx.globalCompositeOperation = 'destination-over'
+        ctx.fillStyle = 'white'
+        ctx.fillRect(0, 0, chartCtx.width, chartCtx.height)
+        ctx.restore()
+      },
+    },
+    // Plugin to display the percentage on top of the bars.
+    {
+      id: 'displayPercentages',
+      afterDraw: (chart) => {
+        const {
+          ctx,
+          data: { datasets },
+        } = chart
+        const _metasets = chart.getDatasetMeta(0)
+        datasets[0].data.forEach((dp, i) => {
+          const barValue = `${percentages[i]}%`
+          const lineHeight = ctx.measureText('M').width
+          ctx.fillStyle = backgroundColors[i]
+          ctx.textAlign = 'center'
+          ctx.fillText(barValue, _metasets.data[i].x, _metasets.data[i].y - lineHeight * 1.5)
+        })
+      },
+    },
+  ]
+
```

Comment:
I would delegate the creation of the above properties to hooks.
something like 
```typescript 
const {downloadChartAsImage} = useActions()
const {plugins} = usePlugins()
```
code above is just an example, organize the properties based on logical grouping if appropriate

Reply:
Plugins are more of a config file that you can add to the Charts. Should we still use hooks if that's the case?